snow/validators/logger.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
+)
+
+var _ NodeStakeLogger = (*logger)(nil)
+
+// NodeStakeLogger allows logging stake changes of a specified validator
+type NodeStakeLogger SetCallbackListener
+
+type logger struct {
+	subnetID ids.ID
+	nodeID   ids.NodeID
+	log      logging.Logger
+}
+
+func NewLogger(subnetID ids.ID, targetNode ids.NodeID, log logging.Logger) NodeStakeLogger {
+	return &logger{
+		subnetID: subnetID,
+		nodeID:   targetNode,
+		log:      log,
+	}
+}
+
+func (l *logger) OnValidatorAdded(valID ids.NodeID, _ *bls.PublicKey, txID ids.ID, weight uint64) {
+	if valID == l.nodeID {
+		l.log.Info("tracking validator: validator added to validators set",
```

Comment:
I'm not sure we really need the `tracking validator:` prefix... We already kinda get that from the stacktrace.

Reply:
The idea was to try and simplify grepping. The common prefix should return only and all logger events.